File: SW_Expert_Academy/6485_삼성시의버스노선.cpp
#include <iostream>
#include <cstring>
using namespace std;
#define f(a,b,c) for(int a=b; a<c;++a)

int main(void) {
    ios_base::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    int T,n,fr,to,p,tmp,max_val; cin >> T;
    int arr[5003];
    f(t,1,T+1) {
        max_val = -1;
        cin >> n;
        f(i,0,n) {
            cin >> fr >> to;
            arr[fr]++;
            arr[to+1]--;
        }
        f(i,1,5001) arr[i]+=arr[i-1];
        cout << '#' << t;
        cin >> p;
        f(i,0,p) {
            cin >> tmp;
            cout<<' '<<arr[tmp];
        }
        cout<<'\n';
        memset(arr,0,sizeof(arr));
    }   
}

File: SW_Expert_Academy/6485_삼성시의버스노선.py
T = int(input())
for t in range(1,T+1):
    n = int(input())
    arr = [0] * 5002
    q = []
    max_val = 0
    for _ in range(n):
        fr, to = map(int,input().split())
        for idx in range(fr,to+1):
            arr[idx]+=1
    p = int(input())
    for _ in range(p):
        q.append(arr[int(input())])
    print(f'#{t}',*q)